app/scripts/modules/core/src/pipeline/config/templates/v2/PipelineTemplatesV2.tsx

```diff
@@ -0,0 +1,121 @@
+import * as React from 'react';
+import { DateTime } from 'luxon';
+import { IPipelineTemplateV2 } from 'core/domain/IPipelineTemplateV2';
+import { ReactModal } from 'core/presentation';
+import {
+  ShowPipelineTemplateJsonModal,
+  IShowPipelineTemplateJsonModalProps,
+} from 'core/pipeline/config/actions/templateJson/ShowPipelineTemplateJsonModal';
+import { PipelineTemplateReader } from '../PipelineTemplateReader';
+
+import './PipelineTemplatesV2.less';
+
+export interface IPipelineTemplatesV2State {
+  templates: IPipelineTemplateV2[];
+  fetchError: Error;
+}
+
+export const PipelineTemplatesV2Error = (props: { message: string }) => {
+  return (
+    <div className="pipeline_templates_error_banner horizontal middle center heading-4">
+      <i className="fa fa-exclamation-triangle" />
+      <span>{props.message}</span>
+    </div>
+  );
+};
+
+export class PipelineTemplatesV2 extends React.Component<{}, IPipelineTemplatesV2State> {
+  constructor(props: {}) {
+    super(props);
+    this.state = { templates: [], fetchError: null };
+  }
+
+  public componentDidMount() {
+    this.fetchTemplates();
+  }
+
+  private fetchTemplates = () => {
+    const templatesPromise = PipelineTemplateReader.getV2PipelineTemplateList();
+    templatesPromise.then(
+      templates => {
+        this.setState({ templates });
+      },
+      err => {
+        this.setState({ fetchError: err });
+      },
+    );
+  };
+
+  private idForTemplate = (template: IPipelineTemplateV2) => {
+    const { id, version = '', digest = '' } = template;
+    return `${id}:${version}:${digest}`;
+  };
+
+  private getUpdateTimeForTemplate = (template: IPipelineTemplateV2) => {
+    const millis = Number.parseInt(template.updateTs, 10);
+    if (isNaN(millis)) {
+      return '';
+    }
+    const dt = DateTime.fromMillis(millis);
+    return dt.toLocaleString(DateTime.DATETIME_SHORT);
+  };
+
+  private showTemplateJson = (template: IPipelineTemplateV2) => {
+    const props = { template, editable: false };
+    ReactModal.show<IShowPipelineTemplateJsonModalProps>(ShowPipelineTemplateJsonModal, props, {
+      dialogClassName: 'modal-lg modal-fullscreen',
+    });
+  };
+
+  public render() {
+    return (
+      <>
+        <div className="infrastructure">
+          <div className="infrastructure-section search-header">
+            <div className="container">
+              <h2 className="header-section">
+                <span className="search-label">Pipeline Templates</span>
+              </h2>
+            </div>
+          </div>
+          <div className="infrastructure-section">
+            <div className="container">
+              {this.state.fetchError && (
+                <PipelineTemplatesV2Error
+                  message={`There was an error fetching pipeline templates: ${this.state.fetchError.message}`}
```

Comment:
If I make the request fail with a 404 or by blocking the request `message` seems to come back `undefined`. In the case of a 404, the `err` object does have a `data.message` and `data.error`